kikit/fab/neodenyy1.py

```diff
@@ -0,0 +1,164 @@
+            continue
+        reference = getReference(c)
+        if reference.startswith("#PWR") or reference.startswith("#FL"):
+            continue
+        if reference in ignore:
+            continue
+        if hasattr(c, "in_bom") and not c.in_bom:
+            continue
+        if hasattr(c, "on_board") and not c.on_board:
+            continue
+        if hasattr(c, "dnp") and c.dnp:
+            continue
+        orderCode = None
+        for fieldName in ordercodeFields:
+            orderCode = getField(c, fieldName)
+            if orderCode is not None and orderCode.strip() != "":
+                break
+        cType = (
+            getField(c, "Value"),
+            getField(c, "Footprint")
+        )
+        bom[cType] = bom.get(cType, []) + [reference]
+    return bom
+
+def noFilter(footprint):
+    return True
+
+def transcodeFootprint(footprint):
+    for f in footprintRe:
+        matchedFootprint = f[0].match(footprint)
+        if matchedFootprint != None:
+            return f[1].format(matchedFootprint.groups()[0])
+    matchedFootprint = footprint.split(':')
+    if len(matchedFootprint) > 1:
+        return matchedFootprint[1].split('_')[0]
+    else:
+        return footprint
+
+def posDataProcess(posData, pcbSize, bom):
+    topLayer = []
+    bottomLayer = []
+    ref = {}
+    for cType, references in bom.items():
+        sortedReferences = sorted(references, key=naturalComponentKey)
+        for refComponent in sortedReferences:
+            ref[refComponent] = cType
+    for line in posData:
+        if line[0] in ref:
+            value, footprint = ref[line[0]]
+            if line[3] == 'T':
+                topLayer.append(line + (value, footprint, line[1], line[2],))
+            elif line[3] == 'B':
+                bottomLayer.append(line + (value, footprint, pcbSize[1] - line[1], line[2], ))
+        else:
+            value = None
+            footprint = None
+            if line[3] == 'T':
+                topLayer.append(line + (value, footprint, line[1], line[2],))
+            elif line[3] == 'B':
+                bottomLayer.append(line + (value, footprint, pcbSize[1] -  line[1], line[2],))
+    return (topLayer, bottomLayer)
+        
+def posDataToCSV(layerData, prepend, filename):
+    basename = os.path.basename(filename)
+    basename = prepend + '_' + basename
+    dirname = os.path.dirname(filename)
+    with open(os.path.join(dirname, basename), "w", newline="", encoding="utf-8") as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerow(["NEODEN","YY1","P&P FILE","","","","","","","","","","",""])
+        writer.writerow(["","","","","","","","","","","","","",""])
+        writer.writerow(["PanelizedPCB","UnitLength","0","UnitWidth","0","Rows","1","Columns","1",""])
+        writer.writerow(["","","","","","","","","","","","","",""])
+        writer.writerow(["Fiducial","1-X","0","1-Y","0","OverallOffsetX","0.00","OverallOffsetY","0.00",""])
```

Comment:
I have zero understanding of the file format; could you confirm that the fiducial positions aren't hard-coded?

Reply:
Add file structure explanation. fiducial position is set to 0,0 machine will disable this feature. If user need it just set it on the machine.